bitnami/prometheus-operator/README.md

```diff
@@ -237,65 +237,66 @@ The following table lists the configurable parameters of the Prometheus Operator
+| `alertmanager.nodeAffinity`                     | Node Affinity (this value is evaluated as a template)                                                                          | `{}`                                                                                                                                                                                                                                                |
+| `alertmanager.podAntiAffinity`                  | Pod anti-affinity policy                                                                                                       | `soft`                                                                                                                                                                                                                                              |
+| `alertmanager.podAffinity`                      | Affinity, in addition to antiAffinity (this value is evaluated as a template)                                                  | `{}`                                                                                                                                                                                                                                                |
+| `alertmanager.nodeSelector`                     | Node labels for pod assignment (this value is evaluated as a template)                                                         | `{}`                                                                                                                                                                                                                                                |
+| `alertmanager.tolerations`                      | List of node taints to tolerate (this value is evaluated as a template)                                                        | `[]`                                                                                                                                                                                                                                                |
+| `alertmanager.config`                           | Alertmanager configuration directive                                                                                           | `{"global":{"resolve_timeout":"5m"},"route":{"group_by":["job"],"group_wait":"30s","group_interval":"5m","repeat_interval":"12h","receiver":"null","routes":[{"match":{"alertname":"Watchdog"},"receiver":"null"}]},"receivers":[{"name":"null"}]}` |
+| `alertmanager.externalConfig`                   | Alertmanager configuration is created externally. If true, `alertmanager.config` is ignored, and a secret will not be created. | `false` See [docs](https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/alerting.md#alerting) for details                                                                                                            |
```

Comment:
With a property named 'existingSecret', you could directly indicate the name of Secret the Chart may use to read the confidential information.

Reply:
I do like that for a user. Unfortunately, alertmanager forces the secret to use the name: `alertmanager-<alertmanager_name>`.
That's why I went with a bool. Since the name is fixed, and is templatized.

Reply:
I do like that for a user. Unfortunately, prometheus-operator forces the secret to use the name: alertmanager-<alertmanager_name>. https://github.com/coreos/prometheus-operator/blob/master/Documentation/user-guides/alerting.md#alerting
That's why I went with a bool. Since the name is fixed, and is templatized.

Reply:
Yes, you are right! Prometheus-operator enforces some naming rules over the Secrets (_alertmanager-<alertmanager_name>_ as you mentioned). Let's analyse that pattern:

**alertmanager-**
This part is always fixed, as it is imposed by Prometheus' conventions.

**<alertmanager_name>**
If we take a look to our `alertmanager/alertmanager.yaml` we find that the `name` uses the following template function:

         `name: {{ template "prometheus-operator.alertmanager.fullname" . }}`

This template uses the release name. Thus, although the Secret must always start with the _alermanager_ prefix, the name is not fixed and depends on the specific release.

That's why I think a string (as in the examples I have shown you) may be more convenient. Thanks!

Reply:
The secret name isn't used anywhere in the chart. The boolean `externalConfig` just configures whether the chart will create secret itself or not.

externalConfig: true -> chart creates the secret, and names it `alertmanager-{{ template "prometheus-operator.alertmanager.fullname" . }}`

externalConfig: false -> chart does nothing. The user creates the secret themselves, and *must* name it `alertmanager-{{ template "prometheus-operator.alertmanager.fullname" . }}`

Reply:
Hi again @RichieSams ! You are absolutely right! Thanks for your contribution 😄